src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -34,13 +35,13 @@ public ErrorResponseDto handleValidationException(
     @ExceptionHandler(RegistrationException.class)
     public ErrorResponseDto handleRegistrationException(RegistrationException e) {
         return generateExceptionDetails("Registration error occurred", BAD_REQUEST,
-            List.of(e.getMessage()));
+            List.of());
```

Comment:
For so called "expected" exception you can safely return message with details. Don't return exception message for exceptions that you don't expect (like handle all exception)